apps/web/src/data/subgraph/requests/memberSnapshot.ts

```diff
@@ -0,0 +1,38 @@
+import * as Sentry from '@sentry/nextjs'
+
+import { SDK } from 'src/data/subgraph/client'
+import { CHAIN_ID } from 'src/typings'
+
+import { DaoTokenOwner_OrderBy, OrderDirection } from '../sdk.generated'
+
+export type DaoMember = {
+  address: string
+  tokens: number[]
+}
+
+export const memberSnapshotRequest = async (
+  chainId: CHAIN_ID,
+  collectionAddress: string
+): Promise<DaoMember[] | undefined> => {
+  try {
+    const data = await SDK.connect(chainId).daoMembersList({
+      where: {
+        dao: collectionAddress,
+      },
+      first: 1000,
```

Comment:
Ideally this should work with DAOs of any size. You can improve this by querying ownerCount before this call and if the ownerCount is > 1000 create n batches where n = ownerCount / 1000 ie 1200 members will create 2 batches, 3000 will create 3 batches then merge them together at the end of the function.